### DISCUSSION TITLE
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree

### DISCUSSION THREAD

**Discussion by JulesVandenbroeck** [columnflow/columnflow#385] [Q&A] [answered]
Hi everyone,

I have been experiencing the warning below when running any task remote (`--workflow htcondor`). This has resulted in the code in my project not being up to date when ran remote (e.g. when `cf.SelectEvents --workflow htcondor` is ran  it runs an outdated version from before this error occurred). Since I am not familiar where git comes into job submission in `law`, I don't know how this occurred _and how to fix this issue_.

```
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

update:

This was fixed by adding a git commit (no need to push), however how this happened is still unknown to me.

**Comment by riga** [comment, marked as answer]
Hi!

It sounds like there was perhaps an issue in your local analysis repository. Before submitting jobs, a cf task called "BundleRepo" is invoked that creates an archive of the current state of your repository (*), and uploads/copies it to a location where remote jobs can access it.

(*) In the bundling process, the repository is copied to a temporary location where uncommitted changes are added (however, no commit is added to your actual repository), and I guess the error you see comes from this process.

Could you run `law run cf.BundleRepo --replicas 1` and confirm that this is working correctly?

**Reply by JulesVandenbroeck** [reply]
`law run cf.BundleRepo --replicas 1` works without any issues! Thank you for the insight into how the  analysis is accessed by jobs, I am slowly understanding the inner workings of cf :)

**Reply by riga** [reply]
Nice!

I will debug this a bit further though, since jobs should actually not be submitted in case the bundling fails, but I'm glad it works for you now.
